### ISSUE TITLE
Incompatible block pointer types sending 'void (^)(.....

### ISSUE THREAD

**Issue by Sun3** [firebase/FirebaseUI-iOS#7]
When I try to compile the FirebaseUIChat app I receive below error in the Viewcontroller with at below line:

```
[self.dataSource populateCellWithBlock:^void(MessageTableViewCell *__nonnull cell, Message *__nonnull message) {
```

Error:
```
FirebaseUIChat/FirebaseUIChat/ViewController.m:34:42: 
Incompatible block pointer types sending 'void (^)(MessageTableViewCell * __nonnull __strong, Message * __nonnull __strong)' to parameter of type 'void (^ __nonnull)(UITableViewCell * __nonnull __strong, NSObject * __nonnull __strong)'
```

Any thoughts?

**Comment by mcdonamp** [comment]
Which version of XCode are you using? Also, are you using the most up to date version of the repo? It should be:

`void (^ __nonnull)(id __nonnull __strong, id __nonnull __strong)`

from [here](https://github.com/firebase/FirebaseUI-iOS/blob/master/FirebaseUI/XCodeMacros.h#L59), for versions of XCode prior to XCode 7 beta 5 (where `__kindof` was introduced). If you haven't, I'd pull the latest changes and see if it works.

**Comment by spf2** [comment]
I'm seeing the same thing.  Xcode 6.4, latest version of code.

**Comment by spf2** [comment]
A quick workaround to get it to compile:
```
   populateCellWithBlock:^void(UITableViewCell *c, NSObject *o) {
      MessageTableViewCell *cell = (MessageTableViewCell*) c;
      Message *message = (Message*) o;
```

**Comment by mcdonamp** [comment]
Ah, the problem is that the Cocoapod hasn't been updated with the __kindof fixes (in https://github.com/firebase/FirebaseUI-iOS/commit/264f596f610c85856e808fea370633c006f0a187). Let me push the pod and it should work as above.

**Comment by mcdonamp** [comment]
This issue should be fixed with [v0.2.5](https://github.com/firebase/FirebaseUI-iOS/releases/tag/v0.2.5). Just `pod update` and it should work again normally.

**Comment by spf2** [comment]
Yup.  Thanks!
